cmake/CMakeLists.txt

```diff
@@ -61,72 +68,134 @@ endif()
 
 enable_testing()
 
+set(UPB_LIBS)
+set(UPB_INTERFACES)
+
 add_library(port INTERFACE)
-add_library(upb
-  ../upb/decode.c
-  ../upb/decode_internal.h
-  ../upb/encode.c
-  ../upb/msg.c
-  ../upb/msg_internal.h
-  ../upb/table.c
-  ../upb/table_internal.h
-  ../upb/upb.c
-  ../upb/upb_internal.h
-  ../upb/decode.h
-  ../upb/encode.h
-  ../upb/msg.h
-  ../upb/upb.h
-  ../upb/upb.hpp)
-target_link_libraries(upb
+list(APPEND UPB_INTERFACES port)
```

Comment:
port is marked as an `INTERFACE` library because it has no sources, but it is not meant to be exported. Users should not be including `port_def.inc` or `port_undef.inc` directly.

To decide whether an interface should be exported, we should look at the visibility on the `cc_library()` rule in `make_cmakelists.py`.